### PR TITLE
Add support for a generic POD type

### PR DESCRIPTION
**Pull request by ronawho**
This lets you write things like:

```chpl
proc myproc(arg: chpl_anyPOD) {
  writeln("A POD type");
}
```

The advantage of doing this over something like:

```chpl
proc myproc(arg) where isPODType(arg) {
  writeln("A POD type");
}
```

is that the generic version works for promoted cases, but the where
clause version does not. This is in support of extending `unorderedCopy`
to all POD types.

Closes https://github.com/Cray/chapel-private/issues/506

Note that the name `chpl_anyPOD` is being used because we're not ready
to make this a user-facing feature (`isPODType` is similarly no-doc'd)
https://github.com/chapel-lang/chapel/issues/14228 captures this issue.

### CODE REVIEW

**Comment by mppf** [comment]
I think the type in the compiler should be called dtAnyPOD.

I think the user-facing type name for this needs more discussion. I doubt that `POD` is what we want.

**Comment by ronawho** [comment]
> I think the type in the compiler should be called dtAnyPOD.

Why's that? We have things like dtNumeric, not dtAnyNumeric

> I think the user-facing type name for this needs more discussion. I doubt that POD is what we want.

Ok, I'll open an issue for that. Do you have any suggestions I can use to seed a list of options with?

File: test/types/POD/elliot/formalPOD.chpl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+var Ai: [1..1] int;
+var Ar: [1..1] real;
+var Ab: [1..1] bool;
+var Atup: [1..1] 2*int;
+var Arec: [1..1] R;
+checkPOD(Ai);
+checkPOD(Ar);
+checkPOD(Ab);
+checkPOD(Atup);
+checkPOD(Arec);
```
Comment (mppf) [inline test/types/POD/elliot/formalPOD.chpl]: This latter part doesn't seem right. Won't these be not pod and fail to resolve in the test as written?

Comment (ronawho) [reply test/types/POD/elliot/formalPOD.chpl]: These trigger promotion

Comment (mppf) [reply test/types/POD/elliot/formalPOD.chpl]: Oh, right, that being the whole point of the PR... Maybe add a comment in the .chpl for the next clueless chump who reads it ?

**Comment by ronawho** [comment]
>> I think the user-facing type name for this needs more discussion. I doubt that POD is what we want.

> Ok, I'll open an issue for that. Do you have any suggestions I can use to seed a list of options with?

I changed it to `chpl_anyPOD` for the time being and will avoid using that in the documented signature for `unorderedCopy`, but thinking about it some more I don't really have a problem with just using `POD` for now. It matches `isPODType`, which also isn't publically documented so if we decide to change the names I guess I don't see the difference between using `POD` vs `chpl_anyPOD` for now.

**Comment by cassella** [comment]
Do you want to add test cases locking in different flavors of class management?  The existing ispod test covers `MyClass` and `borrowed MyClass` (both true). I get true also for `unmanaged`, but false for `owned` and `shared`, and I don't see any other tests that would lock that in.

How about unions?  :)   I also see `string`: false, `c_string`: true, `void`: true

(FWIW, the existing ispod test manages to pass without the compopts file, or even the ispod.h header.)

**Comment by cassella** [comment]
Also BTW, the comment on `isPODType()` says in part

```
User class types in Chapel are always considered POD types ...
```

which doesn't match or mention the behavior for owned and shared class types.  Though it also says `#pragma "no doc"`

**Comment by ronawho** [comment]
> Do you want to add test cases locking in different flavors of class management? The existing ispod test covers MyClass and borrowed MyClass (both true). I get true also for unmanaged, but false for owned and shared, and I don't see any other tests that would lock that in.

Sure, added tests for those. 

>Also BTW, the comment on isPODType() says in part...

That's an outdated comment. Shared/owned are not POD.

> How about unions? :) I also see string: false, c_string: true, void: true

I added a case for string, but we want to remove c_string (https://github.com/chapel-lang/chapel/issues/14215), unions aren't in great shape and are likely to change, and I'm not sure that `void` makes sense. I guess I would lean towards `void` being POD, but it still seems a little odd.

**Pull request merged**